blockchain/util.go

```diff
@@ -80,22 +78,61 @@ func findLogByTopic(ctx context.Context, client *ethclient.Client, tx *types.Tra
 	return nil, fmt.Errorf("cannot find topic \"%s\"in transaction", topic.Hex())
 }
 
-func waitForTransactionResult(ctx context.Context, client *ethclient.Client, logParsePeriod time.Duration, tx *types.Transaction, topic common.Hash) (*types.Log, error) {
+func waitForTransactionResult(ctx context.Context, client Backend, logParsePeriod time.Duration, tx *types.Transaction, topic common.Hash) (*types.Log, error) {
 	tk := util.NewImmediateTicker(logParsePeriod)
 	defer tk.Stop()
 
 	for {
 		select {
 		case <-tk.C:
-			log, err := findLogByTopic(ctx, client, tx, topic)
+			txReceipt, err := client.TransactionReceipt(ctx, tx.Hash())
 			if err != nil {
 				if err == ethereum.NotFound {
 					break
 				}
 				return nil, err
 			}
+			logs, err := FindLogByTopic(ctx, &Receipt{
+				Status: txReceipt.Status,
+				Logs:   txReceipt.Logs,
+			}, topic)
+			if err != nil {
+				return nil, err
+			}
 
-			return log, err
+			return logs, err
+		case <-ctx.Done():
+			return nil, ctx.Err()
+		}
+	}
+}
+
+func WaitTransactionReceipt(ctx context.Context, client CustomEthereumClient, confirmations int64, logParsePeriod time.Duration, tx *types.Transaction) (*Receipt, error) {
```

Comment:
it's rather unclear from the name of this method - what is the difference with the waitForTransactionResult

Reply:
`WaitTransactionReceipt` use Custom client and waiting correct Receipt after confirmations
In other, `waitTransactionResult` stay this as deprecated stuff.
Thanks i would plan to remove this.

Reply:
diff so strange, code changed only 3 lines